RecoLocalMuon/CSCRecHitD/src/CSCHitFromStripOnly.cc

```diff
@@ -472,11 +469,69 @@ void CSCHitFromStripOnly::findMaxima(const CSCDetId& id) {
           break;
         }
       }
-      theConsecutiveStrips.push_back(numberOfConsecutiveStrips);
-    }
-  }
-}
-//
+
+      bool additional_maxima_found = false;
+      // search for additional maxima if:
+      // - hit is closer than 3 strips from the edge
+      // - enough consecutive strips with signal
+      // - strip charge distribution looks abnormal
+
+      if (i > 2 && i + 3 < thePulseHeightMap.size() && numberOfConsecutiveStrips > 3) {
+        //try to look for additional maxima at the left side from the main maxima
+
+        if (((thePulseHeightMap[i + 1].phmax() >= thePulseHeightMap[i - 1].phmax() &&
+              thePulseHeightMap[i + 1].phmax() >= thePulseHeightMap[i - 2].phmax() &&
+              thePulseHeightMap[i + 2].phmax() <= thePulseHeightMap[i - 2].phmax()) ||
+             (thePulseHeightMap[i + 1].phmax() <= thePulseHeightMap[i - 1].phmax() &&
+              thePulseHeightMap[i + 1].phmax() <= thePulseHeightMap[i - 2].phmax())) &&
+            thePulseHeightMap[i - 1].phmax() >=
+                thePulseHeightMap[i - 2]
+                    .phmax() &&  //to avoid close maxima delimitation (this is already present in the code)
```

Comment:
```suggestion
            //to avoid close maxima delimitation (this is already present in the code)
            thePulseHeightMap[i - 1].phmax() >= thePulseHeightMap[i - 2].phmax() &&
```
to improve readability (unfortunately the clang-format breaks up the lines with code and comments; placing them on separate lines is to avoid the issue)